src/cdk-experimental/listbox/listbox.spec.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  ComponentFixture,
+  async,
+  TestBed,
+} from '@angular/core/testing';
+import {Component, DebugElement} from '@angular/core';
+import {By} from '@angular/platform-browser';
+import {
+  CdkOption,
+  CdkListboxModule, CdkListbox
+} from './index';
+import {dispatchMouseEvent} from '@angular/cdk/testing/private';
+
+describe('CdkOption', () => {
+
+  describe('selection state change', () => {
+    let fixture: ComponentFixture<ListboxWithCdkOptions>;
+    let listbox: DebugElement;
+    let listboxInstance: CdkListbox;
+    let options: DebugElement[];
+    let optionInstances: CdkOption[];
+    let optionElements: Element[];
+
+    beforeEach(async(() => {
+      TestBed.configureTestingModule({
+        imports: [CdkListboxModule],
+        declarations: [ListboxWithCdkOptions],
+      }).compileComponents();
+    }));
+
+    beforeEach(async(() => {
+      fixture = TestBed.createComponent(ListboxWithCdkOptions);
+      fixture.detectChanges();
+
+      listbox = fixture.debugElement.query(By.directive(CdkListbox));
+      listboxInstance = listbox.injector.get<CdkListbox>(CdkListbox);
+
+      options = fixture.debugElement.queryAll(By.directive(CdkOption));
+      optionInstances = options.map(o => o.injector.get<CdkOption>(CdkOption));
+      optionElements = options.map(o => o.nativeElement);
+    }));
+
+    it('should generate a unique optionId for each option', () => {
+      let optionIds: string[] = [];
+      for (const instance of optionInstances) {
+        const id = instance._optionid;
+
+        expect(optionIds.indexOf(id)).toBe(-1);
+        optionIds.push(id);
+
+        expect(id).toMatch(/cdk-option-\d+/);
+      }
+    });
+
+    it('should have set the selected input of the options to null by default', () => {
+      for (const instance of optionInstances) {
+        expect(instance.selected).toBeFalse();
+      }
+    });
+
+    it('should update aria-selected when selected is changed programmatically', () => {
+      expect(optionElements[0].getAttribute('aria-selected')).toBeNull();
+      optionInstances[1].selected = true;
+      fixture.detectChanges();
+
+      expect(optionElements[1].getAttribute('aria-selected')).toBe('true');
+    });
+
+    it('should update selected option on click event', () => {
+      let selectedOptions = optionInstances.filter(option => option.selected);
+      spyOn(listboxInstance, '_emitChangeEvent');
```

Comment:
Rather than setting up a spy on the internal listbox method, you can have your `ListboxWithCdkOptions ` component below add a `(selectionChange)` handler to the listbox and assert that that's called. Generally you want to avoid having the test know about the internals of the component unless there's no other practical way to test something.

Reply:
Ok I see. Will change.